app/controllers/candidate_interface/sign_in_controller.rb

```diff
@@ -4,7 +4,21 @@ class SignInController < CandidateInterfaceController
     before_action :redirect_to_application_if_signed_in, except: :authenticate
 
     def new
-      @candidate = Candidate.new
+      if FeatureFlag.active?('improved_expired_token_flow') && params[:u]
+        begin
+          decrypted_candidate_id = Encryptor.decrypt(params[:u])
+
+          candidate = Candidate.find(decrypted_candidate_id)
+          MagicLinkSignIn.call(candidate: candidate)
```

Comment:
We can't do this automatically here without user input. Some email clients follow links, so merely _opening_ an email would trigger a new email with a sign-in link. The user should press a button to explitly request a new sign in link.

Reply:
I've updated this now to add a button and change the content to reflect this with the help of Emma HF and Paul H.